commit 28ceb42daa10d5a209d526a04a0e6767da1621dd
Author: Emelia Smith <ThisIsMissEm@users.noreply.github.com>
Date:   Mon Jul 8 20:57:57 2024 +0200

    Remove OAuth Password Grant Type support

diff --git b/config/initializers/doorkeeper.rb a/config/initializers/doorkeeper.rb
index 5b83685f8f..83100b1cf5 100644
--- b/config/initializers/doorkeeper.rb
+++ a/config/initializers/doorkeeper.rb
@@ -9,9 +9,16 @@ Doorkeeper.configure do
     current_user || redirect_to(new_user_session_url)
   end
 
-  # Disable Resource Owner Password Credentials Grant Flow
-  resource_owner_from_credentials do
-    nil
+  resource_owner_from_credentials do |_routes|
+    user   = User.authenticate_with_ldap(email: request.params[:username], password: request.params[:password]) if Devise.ldap_authentication
+    user ||= User.authenticate_with_pam(email: request.params[:username], password: request.params[:password]) if Devise.pam_authentication
+
+    if user.nil?
+      user = User.find_by(email: request.params[:username])
+      user = nil unless user&.valid_password?(request.params[:password])
+    end
+
+    user unless user&.otp_required_for_login?
   end
 
   # Doorkeeper provides some administrative interfaces for managing OAuth
@@ -163,7 +170,7 @@ Doorkeeper.configure do
   #   http://tools.ietf.org/html/rfc6819#section-4.4.3
   #
 
-  grant_flows %w(authorization_code client_credentials)
+  grant_flows %w(authorization_code password client_credentials)
 
   # Under some circumstances you might want to have applications auto-approved,
   # so that the user skips the authorization step.
